File: iPad/VerBank-CSTSv3-IOSClientApi/ClientStation/api/event/temp/API_IDEvent_Enquirer.m
////
////  API_IDEvent_Enquirer.m
////  JEDIv7-CSTSv3-ClientAPI
////
////  Created by baolin on 13/5/14.
////  Copyright (c) 2013年 wangyubo. All rights reserved.
////
//
//#import "API_IDEvent_Enquirer.h"
//
//@implementation API_IDEvent_Enquirer
//
//-(id)init:(API_IDEventListener *)_listener{
//    if(self=[super init]){
//         listener=_listener;
//        enquiredEventNameSet=[[NSMutableArray alloc]init];
//    }
//    return self;
//}
//
//-(API_IDEventListener *)getListener{
//    return listener;
//}
//
//-(void)addEventName:(NSString *)eventName{
//    @synchronized(enquiredEventNameSet){
//        //[enquiredEventNameSet setValue:eventName];
//    }
//}
//
//-(void)removeEventName:(NSString *)eventName
//{
//    @synchronized(enquiredEventNameSet){
//        [enquiredEventNameSet removeObserver:eventName forKeyPath:nil];
//    }
//}
//
//-(Boolean)containsEvent:(NSString *)eventName{
//    @synchronized(enquiredEventNameSet){
//        return [enquiredEventNameSet containsObject:eventName];
//    }
//}
//
//-(int)eventSize{
//    @synchronized(enquiredEventNameSet){
//        return (int)[enquiredEventNameSet count];
//    }
//}
//-(NSArray *)getEventNameList{
//    @synchronized(enquiredEventNameSet){
//        NSArray *list=[[NSArray alloc]init];
//        for (int i=0; i<[enquiredEventNameSet count]; i++) {
//            //[list setValue:[enquiredEventNameSet objectAtIndex:i]];
//        }
//        return list;
//    }
//}
//
//
//@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/event/API_IDEvent_Enquirer.java
package jedi.ex01.client.station.api.event;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Iterator;
import java.util.List;

public class API_IDEvent_Enquirer {
	private API_IDEventListener listener;
	private HashSet<String> enquiredEventNameSet = new HashSet<String>();

	public API_IDEvent_Enquirer(API_IDEventListener listener) {
		this.listener = listener;
	}

	public API_IDEventListener getListener() {
		return listener;
	}

	public void addEventName(String eventName) {
		synchronized (enquiredEventNameSet) {
			this.enquiredEventNameSet.add(eventName);
		}
	}

	public void removeEventName(String eventName) {
		synchronized (enquiredEventNameSet) {
			this.enquiredEventNameSet.remove(eventName);
		}
	}

	public boolean containsEvent(String eventName) {
		synchronized (enquiredEventNameSet) {
			return enquiredEventNameSet.contains(eventName);
		}
	}

	public int eventSize() {
		synchronized (enquiredEventNameSet) {
			return enquiredEventNameSet.size();
		}
	}

	public List<String> getEventNameList() {
		synchronized (enquiredEventNameSet) {
			ArrayList<String> list = new ArrayList<String>(enquiredEventNameSet
					.size());
			Iterator<String> it = enquiredEventNameSet.iterator();
			while (it.hasNext()) {
				list.add(it.next());
			}
			return list;
		}
	}
}
